integration/app/app.component.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { ɵivyEnabled as ivyEnabled, Component, Directive } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { Subject, of } from 'rxjs';
+import { finalize } from 'rxjs/operators';
+
+import { UntilDestroy } from '@ngneat/until-destroy';
+
+describe('until-destroy runtime behavior', () => {
+  it('ivy has to be enabled', () => {
+    // This assertion has to be performed as we have to
+    // be sure that we're running these tests with the Ivy engine
+    expect(ivyEnabled).toBeTruthy();
+  });
+
+  it('should unsubscribe from the component property', () => {
+    // Arrange
+    @UntilDestroy({ checkProperties: true })
+    @Component({ template: '' })
+    class MockComponent {
+      disposed = false;
+
+      subscription = new Subject()
+        .pipe(
+          finalize(() => {
+            this.disposed = true;
+          })
+        )
+        .subscribe();
+    }
+
+    // Act
+    TestBed.configureTestingModule({
+      declarations: [MockComponent]
+    });
+
+    const fixture = TestBed.createComponent(MockComponent);
+
+    // Assert
+    expect(fixture.componentInstance.disposed).toBeFalsy();
+    fixture.destroy();
+    expect(fixture.componentInstance.disposed).toBeTruthy();
+  });
+
+  it('should unsubscribe from the directive property', () => {
+    // Arrange
+    @Component({
+      template: '<div test></div>'
+    })
+    class MockComponent {}
+
+    @UntilDestroy({ checkProperties: true })
+    @Directive({ selector: '[test]' })
+    class MockDirective {
+      disposed = false;
+
+      subscription = of(10)
```

Comment:
`of` isn't the "best" example as it fires once and completes.

Reply:
Y, I forgot to change to subject 😝 
Will do

Reply:
Done.

Reply:
Forgot to add a test for `untilDestroyed`.